src/details/ArborX_DetailsExpandHalfToFull.hpp

```diff
@@ -0,0 +1,67 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#ifndef ARBORX_DETAILS_EXPAND_HALF_TO_FULL_HPP
+#define ARBORX_DETAILS_EXPAND_HALF_TO_FULL_HPP
+
+#include <ArborX_DetailsKokkosExtViewHelpers.hpp>
+#include <ArborX_DetailsUtils.hpp>
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX::Details
+{
+
+template <class ExecutionSpace, class Offsets, class Indices>
+void expandHalfToFull(ExecutionSpace const &space, Offsets &offsets,
+                      Indices &indices)
+{
+  Kokkos::Profiling::pushRegion("ArborX::Experimental::HalfToFull");
+  typename Offsets::const_type const offsets_orig = offsets;
+  typename Indices::const_type const indices_orig = indices;
+  assert(KokkosExt::lastElement(space, offsets_orig) ==
+         indices_orig.extent_int(0));
+
+  auto const n = offsets.extent(0) - 1;
+  offsets = KokkosExt::cloneWithoutInitializingNorCopying(space, offsets_orig);
```

Comment:
Won't this invalidate all other variables outside of the function that point to that storage? I think we need to use `realloc` here, though it would require us to do full `clone` and not `cloneWithoutInitializingNorCopying` for `_orig` variables.

Reply:
You'd have the same issue with Kokkos::realloc